config/config.go

```diff
@@ -40,16 +43,92 @@ func (e MissingFieldError) Error() string {
 	return fmt.Sprintf("field missing or unset: %s", e.Field)
 }
 
+// DuplicateNameError is an error that includes the duplicate name.
+type DuplicateNameError struct {
+	Name string
+}
+
+func (e DuplicateNameError) Error() string {
+	return fmt.Sprintf("found duplicate name: %s", e.Name)
+}
+
+// Normalize lowercases, and removes all non-alphanumeric characters from a string.
+func Normalize(s string) string {
+	regex := regexp.MustCompile("[^a-zA-Z0-9]+")
+	s = regex.ReplaceAllString(s, "")
+	s = strings.ToLower(s)
+	return s
+}
+
+// ValidateUnique checks that a list has no duplicate entries.
+func ValidateUnique(items []string) error {
+	mErr := &multierror.Error{}
+	set := sets.String{}
```

Comment:
Why don't you just use a dictionary and not bloat the code?
```golang
var errors []error
foo := map[string]bool{}
if _, present := foo[s]; present {
  errors = append(errors, DuplicateNameError{s})
}
foo[s] = true
return fmt.Errorf("%d errors: %s")
```

Reply:
Replaced use of sets, I think multierror should be okay and that the convenience methods are worth it.